src/frontend/apps/desk/src/features/teams/api/useTeam.tsx

```diff
@@ -0,0 +1,35 @@
+import { UseQueryOptions, useQuery } from '@tanstack/react-query';
+
+import { fetchAPI } from '@/api';
+
+import { TeamResponse } from './types';
+
+export type TeamParams = {
+  id: string;
+};
+
+export interface TeamResponseError {
+  detail: string;
+}
```

Comment:
Might be great to declare a generic type/interface `GenericError`. As for now, `TeamResponseError` and `TeamsResponseError` shared the same structure. It won't introduce too much coupling, as we could overload the generic type for our needs.